### ISSUE TITLE
mocking Laravel5 models is problematic due to __mocked being set in Stub::make

### ISSUE THREAD

**Issue by constb** [Codeception/Codeception#1785]
hello. I'm trying to mock models in laravel5 application. I replace ``getAttribute`` and ``setAttribute`` to run assertions against which attributes tested class touches. Unfortunately, codeception stubs also always require me to make an extra exception for ``__mocked`` attribute. Because it is being set after methods are already replaced and there's no ``__mocked`` in model object.

this is minimal example of a helper I use in my tests:

```php
private function fakeUserStub(&$fakeId, &$fakeLastSeen)
{
    $mocked = null; // NEED THIS TO PASS TESTS

    return Stub::make(
        'App\User',
        [
            'getAttribute' => function ($key) use ($fakeId, $fakeLastSeen, $mocked) {

                if ($key === '__mocked') { // NEED THIS TO PASS TESTS
                    return $mocked;        // NEED THIS TO PASS TESTS
                }                          // NEED THIS TO PASS TESTS

                $this->tester->assertContains($key, [ 'id', 'last_seen' ]);

                return ${'fake' . studly_case($key)};
            },
            'setAttribute' => function ($key, $value) use (&$fakeId, &$fakeLastSeen, &$mocked) {

                if ($key === '__mocked') { // NEED THIS TO PASS TESTS
                    $mocked = $value;      // NEED THIS TO PASS TESTS
                    return;                // NEED THIS TO PASS TESTS
                }                          // NEED THIS TO PASS TESTS

                $this->tester->assertContains($key, [ 'id', 'last_seen' ]);

                ${'fake' . studly_case($key)} = $value;
            },
        ]
    );
}
```

**Comment by DavertMik** [comment]
I think `__mocked` can be removed. Actually not sure, but probably it is not used anymore

**Comment by EVODelavega** [comment]
@DavertMik I've encountered some issues with this `__mocked` property, too, when trying to test code that depended on the Couchbase extension (brett19's repo, CouchbaseBucket was the class that I needed). This class implemented the `__set` method, and triggered errors whenever the user tried to set an invalid property. Looking at the Laravel source code, they seem to implement the `__set` method, too, which might explain this bug. So for what it's worth:

I've created a feature branch, and comitted a quick workaround (https://github.com/EVODelavega/Codeception/tree/feature/make-overloading), adding `Stub::makeNative`. If needs be, I'll work on it some more (perhaps use the reflection class to work out if a `__set` method was implemented, and skip the `__mocked` bit in those cases)?
